async/ssl_io.dummy.ml

```diff
@@ -30,10 +30,10 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *---------------------------------------------------------------------------*)
 
-type descriptor = [ `Ssl_not_available ]
-
 open Async
 
+type descriptor = Reader.t * Writer.t * unit Deferred.t
```

Comment:
When it was the old way, it didn't help me to figure out that I didn't have tls-async installed. It was giving some different inscrutable type error and I spent hours trying to figure it out; it would've been nice if it said "this expression is blah but `Tls_not_available` is expected", but it wasn't doing that.

I thought it'd be better the current way, because at least users will get a runtime exception.

Reply:
That's indeed the tradeoff I chose to make: failing at compile time rather than at runtime. 

I'd prefer to see work on a better error message rather than choosing the other axis of the tradeoff.